### PR TITLE
Integrate typescript parser with eslint

### PR DESCRIPTION
**Pull request by henri-hulski**
This PR introduce typescript linting together with eslint.
In detail it do:
- upgrade prettier and eslint/typescript related dependencies
- set the strictPropertyInitialization flag to false to be compatible with previous typescript version
- normalize line endings
- integrate [TypeScript ESLint Parser](https://github.com/eslint/typescript-eslint-parser)
- lint javascript and typescript files with new configuration

### CODE REVIEW

**Comment by henri-hulski** [comment]
Some linting errors (about 18) has still to be fixed or disabled in the file.
Not sure what to do with them so need some help here.

**Comment by henri-hulski** [comment]
Ok I think this PR is ready for review.

File: package.json [hunk 1]
```diff
@@ -64,7 +63,7 @@
     "uuid": "3.1.0"
   },
   "lint-staged": {
-    "*.js": [
+    "*.{js,ts,tsx}": [
```
Comment (Guria) [inline package.json]: btw, what about `md`?

File: packages/demos/todomvc-fluent/src/module/sequences.ts [hunk 2]
```diff
@@ -1,65 +1,64 @@
...
-  // })
-  .paths({
-    true: s => s.action(actions.updateTodoTitle),
-    false: s => s
-  })
-);
+export const submitTodoTitle = sequenceWithProps<{ uid: string }>(s =>
+  s
+    .branch(actions.whenEditedTitle)
+    // .branch<{ true: {}, false: {} }>(x => {
```
Comment (Guria) [inline packages/demos/todomvc-fluent/src/module/sequences.ts]: do not relate to PR, but what about this commented block? cc @christianalfoni

File: packages/node_modules/@cerebral/angular/test/TestComponent.ts [hunk 3]
```diff
@@ -1,34 +1,36 @@
...
-        super({_view:"view"}, {flush:() => {}}); // Test only code normally see below to setup the component
-    }
+  constructor() {
+    super({ _view: 'view' }, { flush: () => {} }) // Test only code normally see below to setup the component
+  }
 
-    // constructor(cdr: ChangeDetectorRef, controller: ControllerService) {
-    //     super(cdr, controller)
-    // }
+  // constructor(cdr: ChangeDetectorRef, controller: ControllerService) {
```
Comment (Guria) [inline packages/node_modules/@cerebral/angular/test/TestComponent.ts]: commented code again

File: packages/node_modules/@cerebral/angular/test/TestComponent.ts [hunk 4]
```diff
@@ -1,34 +1,36 @@
+/* eslint-disable no-unused-vars */
```
Comment (Guria) [inline packages/node_modules/@cerebral/angular/test/TestComponent.ts]: what causes problem here?

File: packages/website/builder/pages/index/index.js [hunk 5]
```diff
@@ -146,8 +146,8 @@ Take a look at a real project using Cerebral, [Codesandbox.io](https://www.codes
             </a>
           </div>
           <div className="index-release">
-            Released under the MIT License
-            Copyright © 2017 Cerebral Github Organisation
+            Released under the MIT License Copyright © 2017 Cerebral Github
```
Comment (Guria) [inline packages/website/builder/pages/index/index.js]: PRobably we need to wrap lines with `<p>` and make year dynamic.

File: package.json (hunk 1 above)
Comment (henri-hulski) [reply package.json]: Maybe next PR. :)

File: packages/node_modules/@cerebral/angular/test/TestComponent.ts (hunk 4 above)
Comment (henri-hulski) [reply packages/node_modules/@cerebral/angular/test/TestComponent.ts]: I think eslint doesn't support decorators.

File: packages/website/builder/pages/index/index.js (hunk 5 above)
Comment (henri-hulski) [reply packages/website/builder/pages/index/index.js]: Fixed. Used <br /> as <p> gave to much space.

File: package.json (hunk 1 above)
Comment (christianalfoni) [reply package.json]: Yeah, let us get this PR done first :)

File: packages/demos/todomvc-fluent/src/module/sequences.ts (hunk 2 above)
Comment (christianalfoni) [reply packages/demos/todomvc-fluent/src/module/sequences.ts]: Removed it now :)

File: packages/node_modules/@cerebral/angular/test/TestComponent.ts (hunk 3 above)
Comment (christianalfoni) [reply packages/node_modules/@cerebral/angular/test/TestComponent.ts]: Removed it :)

**Pull request merged**